### PR TITLE
Do not change last updated status of intra-column-dragged projects

### LINKED ISSUE #1591: Dragging projects always triggered updatedAt
When a project gets dragged, it should not trigger a new `updatedAt` if it does not change the status.

### PR DESCRIPTION
**Pull request by dan-f**
Fixes #1591

Also gets rid of the drag cache `\o/`

### CODE REVIEW

File: src/universal/modules/teamDashboard/components/ProjectColumn/ProjectColumnDropZone.js [hunk 1]
```diff
@@ -1,23 +1,28 @@
...
   area: string,
   atmosphere: Object, // TODO: atmosphere needs a type definition
-  dragCache: DragCache,
+  getProjectById: (ProjectID) => Project,
   lastProject: ?Project, // the last project in a column; may be undefined if the column is empty
-  status: string
+  status: Status
+};
+
+type UpdateProjectMutationArgs = {
```
Comment (mattkrick) [inline src/universal/modules/teamDashboard/components/ProjectColumn/ProjectColumnDropZone.js]: oh man i want to so badly import this from our graphql schema!

File: src/universal/modules/userDashboard/containers/UserColumns/UserColumnsContainer.js [hunk 2]
```diff
@@ -24,6 +24,11 @@ class UserColumnsContainer extends Component {
     }
   }
 
+  getProjectById = (projectId) =>
```
Comment (mattkrick) [inline src/universal/modules/userDashboard/containers/UserColumns/UserColumnsContainer.js]: this is a really neat pattern! probably only makes sense for drags, but it'd be interesting to see if we use it elsewhere.

File: src/universal/modules/teamDashboard/components/ProjectColumn/ProjectColumnDropZone.js (hunk 1 above)
Comment (dan-f) [reply src/universal/modules/teamDashboard/components/ProjectColumn/ProjectColumnDropZone.js]: Yeah, same! Happy to do that the next time around :)

File: src/universal/modules/userDashboard/containers/UserColumns/UserColumnsContainer.js (hunk 2 above)
Comment (dan-f) [reply src/universal/modules/userDashboard/containers/UserColumns/UserColumnsContainer.js]: Than you! I was happy when the idea came to me!

**Pull request merged**